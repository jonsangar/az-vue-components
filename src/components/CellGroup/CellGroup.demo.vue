<template lang="pug">
  .cellGroupDemo
    Demo(title="Uso Básico")
      az-cell-group(title="Datos de usuario")
        az-cell(title="Cell title" value="Content")
        az-cell(title="Cell title" value="Content" label="Description")

    Demo.no-margin-bottom(title="Slots")
      az-cell-group.no-margin-bottom
        template(v-slot:title)
          h4 Slot título
        az-cell(title="Cell title" value="Content")
        az-cell(title="Cell title" value="Content" label="Description")
</template>

<script>
export default {};
</script>

<style lang="scss">
.cellGroupDemo {
  min-height: 100vh;
  background-color: color(gray7);

  .demo__content {
    padding: 0;
  }
}
</style>
