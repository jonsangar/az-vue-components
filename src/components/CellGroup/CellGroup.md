---
device: true
---

# Cell Group

### Instalación

```javascript
import { CellGroup } from 'didor';

Vue.use(CellGroup);
```

---

### Uso

#### Uso básico

``` html
<az-cell-group title="Datos de usuario">
  <az-cell title="Cell title" value="Content"/>
  <az-cell title="Cell title" value="Content" label="Description"/>
</az-cell-group>
```

#### Uso de slots

``` html
<az-cell-group>
  <template v-slot:title>
    <h4>Slot título</h4>
  </template>

  <az-cell title="Cell title" value="Content"/>
  <az-cell title="Cell title" value="Content" label="Description"/>
</az-cell-group>
```

---

### API

| Attribute   | Description                                | Type      | Default   |
| ----------- | ------------------------------------------ | --------- | --------- |
| title | Título del grupo | `String` | -- |

### Slots

| Nombre | Descripción                                             |
| ------ | ------------------------------------------------------- |
| title | Título del grupo |
