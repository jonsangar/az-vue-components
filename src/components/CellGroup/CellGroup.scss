.az-cell-group {
  $rootClass: &;

  margin-bottom: simple();
  border-radius: $border-radius;

  &__content {
      border-radius: $border-radius;
  }

}

@include media(palm) {
  .az-cell-group {
    &__content {
      border-top: 1px solid color(border);
      border-bottom: 1px solid color(border);
    }

    &__title {
      padding: quarter() half();
    }
  }
}
