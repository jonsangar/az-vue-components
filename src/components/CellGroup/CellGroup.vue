<template lang="pug">
  .az-cell-group
    .az-cell-group__title(v-if="title || $slots.title")
      slot(name="title")
        | {{ title }}
    .az-cell-group__content
      slot
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: '',
    },
  },
};
</script>

<style src="./CellGroup.scss" lang="scss"></style>
