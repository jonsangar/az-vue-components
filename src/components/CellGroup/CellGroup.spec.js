import { shallowMount, createLocalVue } from '@vue/test-utils';

// Importamos librerías de estilos e iconos
import SvgIcon from 'vue-svgicon'
import CellGroup from './CellGroup.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('CellGroup.vue', () => {
  const build = () => {
    const wrapper = shallowMount(CellGroup, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente CellGroup con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Comprueba que tiene el titulo', async () => {
    const { wrapper } = build();
    // Si actualizamos la propiedad title, añade el título
    await wrapper.setProps({
      title: 'myTitle',
    });
    // Esperamos para que se actualice la prop
    await wrapper.vm.$nextTick()
    // Comprobamos que se ha activado la prop
    expect(wrapper.vm.title).toStrictEqual('myTitle');
    // Comprobamos que se ha añadido el titulo con estilos por defecto
    expect(wrapper.find('.az-cell-group__title').isVisible()).toBe(true);
    expect(wrapper.find('.az-cell-group__title').text()).toBe('myTitle');
  });
})
