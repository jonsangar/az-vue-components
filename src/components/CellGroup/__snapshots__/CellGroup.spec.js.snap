// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellGroup.vue Comprueba renderizado del componente CellGroup con snapshot 1`] = `
<div class="az-cell-group">
  <!---->
  <div class="az-cell-group__content"></div>
</div>
`;
